### PR TITLE
sardana motor

### PR DESCRIPTION
**Pull request by meguiraun**
a bit of cleanup and make it more in line with other Motors.

ping @gabadinho

### CODE REVIEW

File: mxcubecore/HardwareObjects/SardanaMotor.py [hunk 1]
```diff
@@ -20,41 +20,27 @@
...
-        "INIT": SardanaMotorState.UNUSABLE,
-        "RUNNING": SardanaMotorState.UNUSABLE,
-        "ALARM": SardanaMotorState.UNUSABLE,
-        "DISABLE": SardanaMotorState.UNUSABLE,
-        "UNKNOWN": SardanaMotorState.UNUSABLE,
-    }
-
     def __init__(self, name):
         AbstractMotor.__init__(self, name)
         HardwareObject.__init__(self, name)
```
Comment (beteva) [inline mxcubecore/HardwareObjects/SardanaMotor.py]: No problem at all, but why not using super()?

File: mxcubecore/HardwareObjects/SardanaMotor.py [hunk 2]
```diff
@@ -20,41 +20,27 @@
...
+    FAULT = HardwareObjectState.FAULT
+    INIT = HardwareObjectState.BUSY
+    RUNNING = HardwareObjectState.BUSY
+    ALARM = HardwareObjectState.WARNING
+    DISABLE = HardwareObjectState.OFF
+    UNKNOWN = HardwareObjectState.UNKNOWN
+    INVALID = HardwareObjectState.FAULT
 
 
 class SardanaMotor(AbstractMotor, HardwareObject):
```
Comment (beteva) [inline mxcubecore/HardwareObjects/SardanaMotor.py]: HardwareObject is already inherited by AbstractMotor. Do you need it here?

File: mxcubecore/HardwareObjects/SardanaMotor.py [hunk 3]
```diff
@@ -287,11 +243,3 @@ def test_hwo(hwo):
     print(("Position for %s is: %s" % (hwo.username, hwo.get_value())))
     print(("Velocity for %s is: %s" % (hwo.username, hwo.get_velocity())))
     print(("Acceleration for %s is: %s" % (hwo.username, hwo.get_acceleration())))
```
Comment (beteva) [inline mxcubecore/HardwareObjects/SardanaMotor.py]: The linter suggest to use f-strings.

File: mxcubecore/HardwareObjects/SardanaMotor.py [hunk 4]
```diff
@@ -70,7 +56,6 @@ def __init__(self, name):
         self.limit_lower = None
         self.static_limits = (-1e4, 1e4)
         self.limits = (None, None)
-        self.motor_state = SardanaMotorState.NOTINITIALIZED
 
     def init(self):
         super().init()
```
Comment (beteva) [inline mxcubecore/HardwareObjects/SardanaMotor.py:63]: The actuator_name is defined by the inherited classes, so, you can remove the line
`self.actuator_name = self.get_property("actuator_name")`

**Review by beteva — APPROVED** [review]
All looks fine. Few cosmetic comments, as you are in a clean-up mood 😄 
Also for the documentation, it l;looks prettier if you use code-block. Note the blank line above and below the code-block statement and also leave one blank character in front of `<object>` and `</object>`. Also device is obsolete, please use object instead in the xml example.

```
Example xml_ configuration: 

.. code-block:: xml

 <object class="SardanaMotor">
    <taurusname>dmot01</taurusname>
    <username>Dummy</username>
    <actuator_name>dummy_motor</actuator_name>
    <threshold>0.005</threshold>
    <move_threshold>0.005</move_threshold>
    <interval>2000</interval>
 </object>

```

File: mxcubecore/HardwareObjects/SardanaMotor.py (hunk 3 above)
Comment (meguiraun) [reply mxcubecore/HardwareObjects/SardanaMotor.py]: I removed those completely, that test method is useless as it is

**Comment by meguiraun** [comment]
thank, Antonia!

File: mxcubecore/HardwareObjects/SardanaMotor.py [hunk 5]
```diff
@@ -8,56 +8,44 @@
 """
 Interfaces Sardana Motor objects.
 taurusname is the only obligatory property.
-<device class="SardanaMotor">
+
+.. code-block:: xml
+
+ <object class="SardanaMotor">
```
Comment (fabcor-maxiv) [inline mxcubecore/HardwareObjects/SardanaMotor.py]:
```suggestion
   <object class="SardanaMotor">
```

File: mxcubecore/HardwareObjects/SardanaMotor.py [hunk 6]
```diff
@@ -8,56 +8,44 @@
...
     <taurusname>dmot01</taurusname>
     <username>Dummy</username>
     <actuator_name>dummy_motor</actuator_name>
     <threshold>0.005</threshold>
     <move_threshold>0.005</move_threshold>
     <interval>2000</interval>
-</device>
+ </object>
```
Comment (fabcor-maxiv) [inline mxcubecore/HardwareObjects/SardanaMotor.py]:
```suggestion
   </object>
```

File: mxcubecore/HardwareObjects/SardanaMotor.py [hunk 7]
```diff
@@ -8,56 +8,44 @@
 """
 Interfaces Sardana Motor objects.
 taurusname is the only obligatory property.
-<device class="SardanaMotor">
+
```
Comment (fabcor-maxiv) [inline mxcubecore/HardwareObjects/SardanaMotor.py:11]: This whole docstring block is not rendered, not sure why.

File: mxcubecore/HardwareObjects/SardanaMotor.py [hunk 8]
```diff
@@ -150,71 +137,60 @@ def init(self):
...
-        if self.limit_lower is None:
-            self.limit_lower = self.static_limits[0]
-
-        if self.limit_upper is None:
-            self.limit_upper = self.static_limits[1]
+    def get_state(self):
+        """Get the motor state.
+        Returns:
+            (enum 'HardwareObjectState'): Motor state.
+        """
```
Comment (fabcor-maxiv) [inline mxcubecore/HardwareObjects/SardanaMotor.py]:
```suggestion
    def get_state(self) -> HardwareObjectState:
        """Get the motor state

        Returns:
            Motor state
        """
```

File: mxcubecore/HardwareObjects/SardanaMotor.py [hunk 9]
```diff
@@ -150,71 +137,60 @@ def init(self):
...
+    def _update_state(self, state):
+        try:
+            state = state.upper()
+            state = SardanaMotorState[state].value
+        except (AttributeError, KeyError):
+            state = self.STATES.UNKNOWN
+        return self.update_state(state)
 
-    def motor_state_changed(self, state=None):
+    def is_ready(self):
```
Comment (fabcor-maxiv) [inline mxcubecore/HardwareObjects/SardanaMotor.py]:
```suggestion
    def is_ready(self) -> bool:
```

File: mxcubecore/HardwareObjects/SardanaMotor.py [hunk 10]
```diff
@@ -150,71 +137,60 @@ def init(self):
...
         """
-        Descript. : called by the state channels update event
-                    checks if the motor is at it's limit,
-                    and sets the new device state
+        Descript. : True if the motor is ready
```
Comment (fabcor-maxiv) [inline mxcubecore/HardwareObjects/SardanaMotor.py]:
```suggestion
        Returns:
            True if the motor is ready
```

File: mxcubecore/HardwareObjects/SardanaMotor.py [hunk 11]
```diff
@@ -150,71 +137,60 @@ def init(self):
...
-        if motor_state != self.motor_state:
-            self.motor_state = motor_state
-            self.emit("stateChanged", (motor_state,))
+    def wait_ready(self, timeout=None):
+        with Timeout(timeout, RuntimeError("Timeout waiting for status ready")):
+            while not self.is_ready():
+                time.sleep(0.1)
 
-    def motor_position_changed(self, position=None):
+    def is_moving(self):
```
Comment (fabcor-maxiv) [inline mxcubecore/HardwareObjects/SardanaMotor.py]:
```suggestion
    def is_moving(self) -> bool:
```

File: mxcubecore/HardwareObjects/SardanaMotor.py [hunk 12]
```diff
@@ -150,71 +137,60 @@ def init(self):
...
         """
-        Descript. : called by the position channels update event
-                    if the position change exceeds threshold,
-                    valueChanged is fired
+        Descript. : True if the motor is currently moving
```
Comment (fabcor-maxiv) [inline mxcubecore/HardwareObjects/SardanaMotor.py]:
```suggestion
        Returns:
            True if the motor is currently moving
```

File: mxcubecore/HardwareObjects/SardanaMotor.py [hunk 13]
```diff
@@ -150,71 +137,60 @@ def init(self):
...
-        if position is None:
-            position = self.position_channel.get_value()
-        if abs(self.motor_position - position) >= self.threshold:
-            self.motor_position = position
-            self.emit("valueChanged", (position,))
-            self.motor_state_changed()
+        return self.get_state() == HardwareObjectState.BUSY
 
-    def get_state(self):
+    def wait_end_of_move(self, timeout=None):
```
Comment (fabcor-maxiv) [inline mxcubecore/HardwareObjects/SardanaMotor.py]:
```suggestion
    def wait_end_of_move(self, timeout: float = None) -> None:
```

File: mxcubecore/HardwareObjects/SardanaMotor.py [hunk 14]
```diff
@@ -150,71 +137,60 @@ def init(self):
...
         """
-        Descript. : returns the current motor state
+        Descript. : waits till the motor stops
```
Comment (fabcor-maxiv) [inline mxcubecore/HardwareObjects/SardanaMotor.py]:
```suggestion
        Wait till the motor stops
```

**Review by fabcor-maxiv — COMMENTED** [review]
The rendered docstrings can be checked here: https://mxcubecore--946.org.readthedocs.build/en/946/dev/autosummary/mxcubecore.HardwareObjects.SardanaMotor.html

**Comment by gabadinho** [comment]
Hi Mikel, the code looks fine by me, give me a couple of days and I'll test it here in ALBA to check if any issues appear!
Cheers, José

**Comment by marcus-oscarsson** [comment]
@gabadinho, @meguiraun Does it mean that I can merge this in the meantime or is it still sort of WIP ?

**Pull request merged**